android/src/com/mapswithme/maps/base/BaseActivityDelegate.java

```diff
@@ -77,6 +78,7 @@ public void onResume()
   {
     logLifecycleMethod("onResume()");
     ViewServer.get(mActivity.get()).setFocusedWindow(mActivity.get());
+    Utils.showOnLockScreen(Config.isShowOnLockScreenEnabled(), mActivity.get());
```

Comment:
Shouldn't flags be set once in onCreate?

Reply:
I think it's needed.
Such scenario
1. Preference is off
2. Open settings
3. Enable preference
4. Go back to map activity
5. I believe `onCreate` wouldn't be called and therefore flags are in a state before preference change.
As always with questions about Android/Activity lifecycle it's not obvious.
I haven't actually tested whether `onCreate` is called - I think it would be only if the system decided to kill this activity which shouldn't be the case for an active app.